### PR TITLE
feat: ListSubheader component

### PR DESCRIPTION
**Pull request by ocarreterom**
### Motivation
Add new component `ListSubheader` to be use in `FlatList`.
I didn't found specific specs for this component in material.io but there are some examples of lists with subheader. https://material.io/design/components/lists.html#anatomy

closes https://github.com/callstack/react-native-paper/issues/716

### Test plan
|Reference|react-native-paper|
|----------|-------------------|
|![reference]|![paper]|

[reference]: https://user-images.githubusercontent.com/11599942/51439310-8efcc080-1cb8-11e9-9347-3d8ff99856ac.png
[paper]: https://user-images.githubusercontent.com/11599942/53601703-fb6ab980-3bac-11e9-8efc-9d390a61920e.png

### CODE REVIEW

**Comment by ferrannp** [comment]
@ocarreterom can you fix merge conflicts?

File: src/components/List/ListSection.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 import { View, StyleSheet } from 'react-native';
-import Text from '../Typography/Text';
+import ListSubheader from './ListSubheader';
 import { withTheme } from '../../core/theming';
 import type { Theme } from '../../types';
 
 type Props = React.ElementConfig<typeof View> & {
   /**
    * Title text for the section.
+   * @deprecated
```
Comment (ferrannp) [inline src/components/List/ListSection.js]: What's this about?

Comment (ocarreterom) [reply src/components/List/ListSection.js]: I'm trying to avoid a breaking change. 😅
How can I inidicate that `title` prop should not be used?

**Comment by ocarreterom** [comment]
Rebased. 👌

**Comment by Trancever** [comment]
@satya164 Any comments on this? Imo this change is ok and it allows passing `Subheader` to `FlatList`. I am also thinking if we should keep title prop on List.Section, remove it or maybe mark as deprecated and remove while releasing 3.0 version. What do you think?

File: src/components/List/ListSection.js (hunk 1 above)
Comment (Trancever) [inline src/components/List/ListSection.js]: Let's remove this warning, and keep previous prop as it is.

File: src/components/List/ListSubheader.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ * class MyComponent extends React.Component {
+ *  render () {
+ *    return <List.Subheader>My List Title</List.Subheader>;
+ *  }
+ * }
+ *
+ * export default MyComponent;
+ * ```
+ */
+class ListSubheader extends React.Component<Props> {
```
Comment (Trancever) [inline src/components/List/ListSubheader.js]: Why do we call it Subheader? Isn't it simply Header?

Comment (ocarreterom) [reply src/components/List/ListSubheader.js]: The name comes from [material.io](https://material.io/design/components/lists.html#anatomy). 🤷‍♂️

Comment (Trancever) [reply src/components/List/ListSubheader.js]: It's ok then :D

File: src/components/List/ListSection.js (hunk 1 above)
Comment (ocarreterom) [reply src/components/List/ListSection.js]: Done! 👌

File: src/components/List/ListSubheader.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  static displayName = 'List.Subheader';
+
+  render() {
+    const { style, theme, ...rest } = this.props;
+    const { colors, fonts } = theme;
+
+    const computedStyle = StyleSheet.flatten([
+      styles.container,
+      {
+        color: colors.text,
```
Comment (Trancever) [inline src/components/List/ListSubheader.js]: color should be: `color(colors.text).alpha(0.54).rgb().string();` just like before.
Same for font, it should be: `fonts.medium`

File: src/components/List/ListSubheader.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+ * ```
+ */
+class ListSubheader extends React.Component<Props> {
+  static displayName = 'List.Subheader';
+
+  render() {
+    const { style, theme, ...rest } = this.props;
+    const { colors, fonts } = theme;
+
+    const computedStyle = StyleSheet.flatten([
```
Comment (Trancever) [inline src/components/List/ListSubheader.js]: Can we remove this variable and pass array with 3 objects directly to style prop?

File: src/components/List/ListSubheader.js (hunk 3 above)
Comment (Trancever) [reply src/components/List/ListSubheader.js]: Also, let's extract them to separate variables just like before.

File: src/components/List/ListSubheader.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
+/* @flow */
+
+import * as React from 'react';
+import { Text, StyleSheet } from 'react-native';
```
Comment (Trancever) [inline src/components/List/ListSubheader.js]: Text should be imported from `react-native-paper`

File: src/components/List/ListSubheader.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+      style,
+    ]);
+
+    return <Text numberOfLines={1} {...rest} style={computedStyle} />;
+  }
+}
+
+const styles = StyleSheet.create({
+  container: {
+    fontSize: 14,
```
Comment (Trancever) [inline src/components/List/ListSubheader.js]: Use same styles as before 
`{
  marginVertical: 13,
  marginHorizontal: 16
}`

Comment (ocarreterom) [reply src/components/List/ListSubheader.js]: Isn't it better to use padding than margin?
I think that using padding is more easy to apply custom common styles like background color.

Comment (Trancever) [reply src/components/List/ListSubheader.js]: I am ok with paddings, let's change it but we should keep the previous values. When those issues are solved, we will be ready to merge it :)

Comment (Trancever) [reply src/components/List/ListSubheader.js]: I am ok with paddings, let's change it but we should keep the previous values. When those issues are solved, we will be ready to merge it :)

**Comment by ocarreterom** [comment]
Issues addressed 👍

**Pull request merged**